File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hashmap = new HashMap<>();
        int n = nums.length;
        int[] ans = new int[2];
        for (int i = 0; i < n; i++){
            int difference = target - nums[i];
            if (hashmap.containsKey(difference)){
                ans[0] = hashmap.get(difference);
                ans[1] = i;
                return ans;
            }
            hashmap.put(nums[i], i);
        }

        return ans;
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        int n = nums.size();
        vector<int> ans;

        for(int i = 0; i < n; i++){
            int difference = target - nums[i];
            if (hashmap.find(difference) != hashmap.end()){
                ans = {i, hashmap[difference]};
                return ans;
            }
            hashmap[nums[i]] = i;
        }
        return ans;
    }
};